piptools/scripts/sync.py

```diff
@@ -60,15 +68,15 @@ def cli(dry_run, force, find_links, index_url, extra_index_url, no_index, quiet,
     to_install, to_uninstall = sync.diff(requirements, installed_dists)
 
     install_flags = []
-    for link in find_links or []:
+    for link in repository.finder.find_links or []:
         install_flags.extend(['-f', link])
-    if no_index:
```

Comment:
With this change the no_index flag is inhibited. Is this wanted?

Reply:
The `no_index` flag is passed to the `get_pip_options_and_pypi_repository` call which then passes it along to the `finder` in the constructed repository. That'll make the `repository.finder.index_urls` to be empty if `no_index` is `True`. It'll also consider `--no-index` specified in the parsed txt file.